simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,787 @@
+from enum import Enum
+from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.cuds.particles import Particle, Particles
+from simphony.cuds.lattice import Lattice
+import numpy as np
+import subprocess
+#from simphony.cuds.abstractparticles import ABCParticles
+
+import uuid
+import logging
+
+import os.path
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+
+class qe_functions(object):
+    '''
+    functions for reading and writing quantum espresso input and output files
+    '''
+
+    def __init__(self):
+        self.SP = DataContainer()   #System Parameters and Conditions
+#        SP = self.data_container
+        self.pc =  Particles('quantum_espresso_particles')
+
+
+
+    def ReadEspressoOutputFile(self,file_name):
+        '''
+        This function parses  Espresso output files which usually will have a name like 'name.charge'.
+        This file has the structure shown at http://phycomp.technion.ac.il/~sbgrosso/QE_charge_density/node18.html
+        The first line has 8 numbers, the three first numbers are the size of the grid in the three spatial directions,
+        the three following are the same repeated.
+        The seventh number corresponds to the number of atoms and the last one to the number of different type of atoms.
+        Second line - the first number is the type of Bravais lattice,
+        and the three following numbers are the celldimensions defined in the input file of pw.x.
+        The charge density is defined for each point of the grid starting after the atom definitions.
+        :param file_name: name of the espresso output file
+        :return:
+        '''
+
+        if not(os.path.exists(file_name)):
+            logging.debug("file "+str(file_name)+" not found")
+            return(1)
```

Comment:
I think raising an exception should be performed.   Especially when this return code isn't checked anywhere